client/RequestBehavior.ts

```diff
@@ -1,34 +1,39 @@
 import { RequestType } from '../src/lib/RequestTypes';
 import { PostMessageRpcClient, RedirectRpcClient } from '@nimiq/rpc';
+import { ResultByRequestType } from '../src/lib/PublicRequestTypes';
 
-export class RequestBehavior {
+export abstract class RequestBehavior<B extends BehaviorType> {
     public static getAllowedOrigin(endpoint: string) {
         const url = new URL(endpoint);
         return url.origin;
     }
 
-    private readonly _type: BehaviorType;
+    private readonly _type: B;
 
-    constructor(type: BehaviorType) {
+    constructor(type: B) {
         this._type = type;
```

Comment:
Is this used? I can't seem to find it anywhere.

Reply:
Turns out it's not used. But I can't get type magic to work if I delete the whole `BehaviorType`.

Reply:
Could then still be removed as it is unnecessary.